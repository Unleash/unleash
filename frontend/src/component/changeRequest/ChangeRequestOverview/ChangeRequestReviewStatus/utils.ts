export const getBrowserTimezoneInHumanReadableUTCOffset = (
    date = new Date(),
): string => {
    const offset = -date.getTimezoneOffset();
    const hours = Math.floor(Math.abs(offset) / 60);
    const minutes = Math.abs(offset) % 60;
    let sign = '+';
    if (offset < 0) {
        sign = '-';
    }

    // Ensure that hours and minutes are two digits
    const zeroPaddedHours = hours.toString().padStart(2, '0');
    const zeroPaddedMinutes = minutes.toString().padStart(2, '0');

    return `UTC${sign}${zeroPaddedHours}:${zeroPaddedMinutes}`;
};

export const getBrowserTimezone = (): string => {
    return Intl.DateTimeFormat().resolvedOptions().timeZone;
};
