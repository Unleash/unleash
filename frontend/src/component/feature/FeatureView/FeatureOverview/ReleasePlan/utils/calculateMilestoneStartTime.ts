import type { IReleasePlanMilestone } from 'interfaces/releasePlans';
import { addMinutes } from 'date-fns';

const parseStartTime = (startedAt: string | null | undefined): Date | null => {
    if (!startedAt) return null;

    const date = new Date(startedAt);
    if (Number.isNaN(date.getTime())) {
        return null;
    }
    return date;
};

const getIntervalMinutes = (
    milestone: IReleasePlanMilestone,
): number | null => {
    const intervalMinutes = milestone.transitionCondition?.intervalMinutes;

    if (!intervalMinutes) {
        return null;
    }
    return intervalMinutes;
};

const findMostRecentStartedMilestone = (
    milestones: IReleasePlanMilestone[],
    targetIndex: number,
): { index: number; startTime: Date } | null => {
    for (let i = targetIndex; i >= 0; i--) {
        const startTime = parseStartTime(milestones[i].startedAt);
        if (startTime) {
            return { index: i, startTime };
        }
    }
    return null;
};

const findBaselineMilestone = (
    milestones: IReleasePlanMilestone[],
    targetIndex: number,
    activeMilestoneId?: string,
): { index: number; startTime: Date } | null => {
    if (!activeMilestoneId) {
        return findMostRecentStartedMilestone(milestones, targetIndex);
    }

    const activeIndex = milestones.findIndex((m) => m.id === activeMilestoneId);
    if (activeIndex === -1 || activeIndex > targetIndex) {
        return findMostRecentStartedMilestone(milestones, targetIndex);
    }

    const activeStartTime = parseStartTime(milestones[activeIndex].startedAt);
    if (activeStartTime) {
        return { index: activeIndex, startTime: activeStartTime };
    }

    return findMostRecentStartedMilestone(milestones, targetIndex);
};

const calculateTimeFromBaseline = (
    milestones: IReleasePlanMilestone[],
    baseline: { index: number; startTime: Date },
    targetIndex: number,
): Date | null => {
    let currentTime = baseline.startTime;

    for (let i = baseline.index; i < targetIndex; i++) {
        const previousMilestone = milestones[i];
        const intervalMinutes = getIntervalMinutes(previousMilestone);

        if (!intervalMinutes) return null;

        currentTime = addMinutes(currentTime, intervalMinutes);
    }

    return currentTime;
};

export const calculateMilestoneStartTime = (
    milestones: IReleasePlanMilestone[],
    targetMilestoneId: string,
    activeMilestoneId?: string,
): Date | null => {
    const targetIndex = milestones.findIndex((m) => m.id === targetMilestoneId);
    if (targetIndex === -1) return null;

    const baseline = findBaselineMilestone(
        milestones,
        targetIndex,
        activeMilestoneId,
    );
    if (!baseline) return null;

    if (baseline.index === targetIndex) {
        return baseline.startTime;
    }

    return calculateTimeFromBaseline(milestones, baseline, targetIndex);
};
