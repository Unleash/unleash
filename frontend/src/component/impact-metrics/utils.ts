export const getTimeUnit = (selectedRange: string) => {
    switch (selectedRange) {
        case 'hour':
            return 'minute';
        case 'day':
            return 'hour';
        case 'week':
            return 'day';
        case 'month':
            return 'week';
        default:
            return 'day';
    }
};

export const getDisplayFormat = (selectedRange: string) => {
    switch (selectedRange) {
        case 'hour':
            return 'HH:mm';
        case 'day':
            return 'HH:mm';
        case 'week':
            return 'MMM dd';
        case 'month':
            return 'MMM dd';
        default:
            return 'MMM dd';
    }
};

export const getSeriesLabel = (metric: Record<string, string>): string => {
    const { __name__, ...labels } = metric;

    const labelParts = Object.entries(labels)
        .filter(([key, value]) => key !== '__name__' && value)
        .map(([key, value]) => `${key}=${value}`)
        .join(', ');

    if (!__name__ && !labelParts) {
        return 'Series';
    }

    if (!__name__) {
        return labelParts;
    }

    if (!labelParts) {
        return __name__;
    }

    return `${__name__} (${labelParts})`;
};

export const formatLargeNumbers = (value: number): string => {
    if (value >= 1000000) {
        return `${(value / 1000000).toFixed(0)}M`;
    }
    if (value >= 1000) {
        return `${(value / 1000).toFixed(0)}k`;
    }
    return value.toString();
};

export const getMetricType = (seriesName: string) => {
    if (seriesName.startsWith('unleash_counter_')) return 'counter';
    if (seriesName.startsWith('unleash_gauge_')) return 'gauge';
    return 'unknown';
};
