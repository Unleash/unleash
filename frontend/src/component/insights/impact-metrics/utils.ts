export const getTimeUnit = (selectedRange: string) => {
    switch (selectedRange) {
        case 'hour':
            return 'minute';
        case 'day':
            return 'hour';
        case 'week':
            return 'day';
        case 'month':
            return 'day';
        default:
            return 'hour';
    }
};

export const getDisplayFormat = (selectedRange: string) => {
    switch (selectedRange) {
        case 'hour':
        case 'day':
            return 'HH:mm';
        case 'week':
        case 'month':
            return 'MMM dd';
        default:
            return 'MMM dd HH:mm';
    }
};

export const getSeriesLabel = (metric: Record<string, string>): string => {
    const { __name__, ...labels } = metric;

    const labelParts = Object.entries(labels)
        .filter(([key, value]) => key !== '__name__' && value)
        .map(([key, value]) => `${key}=${value}`)
        .join(', ');

    if (!__name__ && !labelParts) {
        return 'Series';
    }

    if (!__name__) {
        return labelParts;
    }

    if (!labelParts) {
        return __name__;
    }

    return `${__name__} (${labelParts})`;
};

export const formatLargeNumbers = (value: number): string => {
    if (value >= 1000000) {
        return `${(value / 1000000).toFixed(0)}M`;
    }
    if (value >= 1000) {
        return `${(value / 1000).toFixed(0)}k`;
    }
    return value.toString();
};
