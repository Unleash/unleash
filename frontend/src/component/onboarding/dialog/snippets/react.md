1\. Install the SDK
```sh
npm install @unleash/proxy-client-react unleash-proxy-client
```

2\. Initialize Unleash
```jsx
import { createRoot } from 'react-dom/client';
import { FlagProvider } from '@unleash/proxy-client-react';

const config = {
  url: '<YOUR_API_URL>',
  clientKey: '<YOUR_API_TOKEN>', // in production use environment variable
  appName: 'unleash-onboarding-react',
};

const root = createRoot(document.getElementById('root'));

root.render(
  <React.StrictMode>
    <FlagProvider config={config}>
      <App />
    </FlagProvider>
  </React.StrictMode>
);
```

3\. Check feature flag status
```jsx
import { useFlag } from '@unleash/proxy-client-react';

const TestComponent = () => {
  const enabled = useFlag('<YOUR_FLAG>');

  return enabled ? 'Flag is enabled' : 'Flag is disabled'
};
```
---
```jsx
const config = {
  url: '<YOUR_API_URL>', 
  clientKey: process.env.UNLEASH_API_TOKEN,
  appName: 'unleash-onboarding-react',
};
```

---
- [SDK repository with documentation](https://github.com/Unleash/proxy-client-react)
- [React SDK example with CodeSandbox](https://github.com/Unleash/unleash-sdk-examples/tree/main/React)
- [https://docs.getunleash.io/feature-flag-tutorials/react](https://docs.getunleash.io/feature-flag-tutorials/react)

