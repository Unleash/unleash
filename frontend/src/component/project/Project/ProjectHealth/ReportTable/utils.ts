import differenceInDays from 'date-fns/differenceInDays';
import { EXPERIMENT, OPERATIONAL, RELEASE } from 'constants/featureToggleTypes';

const FORTY_DAYS = 40;
const SEVEN_DAYS = 7;

export const toggleExpiryByTypeMap: Record<string, number> = {
    [EXPERIMENT]: FORTY_DAYS,
    [RELEASE]: FORTY_DAYS,
    [OPERATIONAL]: SEVEN_DAYS,
};

export const getDiffInDays = (date: Date, now: Date) => {
    return Math.abs(differenceInDays(date, now));
};

export const expired = (diff: number, type: string) => {
    return diff >= toggleExpiryByTypeMap[type];
};
