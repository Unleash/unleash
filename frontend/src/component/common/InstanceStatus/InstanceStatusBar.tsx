import { styled, Button, Typography } from '@mui/material';
import { IInstanceStatus } from 'interfaces/instance';
import { INSTANCE_STATUS_BAR_ID } from 'utils/testIds';
import { InfoOutlined, WarningAmber } from '@mui/icons-material';
import { useNavigate } from 'react-router-dom';
import { useContext } from 'react';
import AccessContext from 'contexts/AccessContext';
import { ADMIN } from 'component/providers/AccessProvider/permissions';
import {
    trialHasExpired,
    trialExpiresSoon,
    isTrialInstance,
} from 'utils/instanceTrial';
import { formatDistanceToNowStrict, parseISO } from 'date-fns';

const StyledWarningBar = styled('aside')(({ theme }) => ({
    position: 'relative',
    zIndex: 1,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    padding: theme.spacing(1),
    gap: theme.spacing(1),
    borderBottom: '1px solid',
    borderColor: theme.palette.warning.border,
    background: theme.palette.warning.light,
    color: theme.palette.warning.dark,
}));

const StyledInfoBar = styled('aside')(({ theme }) => ({
    position: 'relative',
    zIndex: 1,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    padding: theme.spacing(1),
    gap: theme.spacing(1),
    borderBottom: '1px solid',
    borderColor: theme.palette.info.border,
    background: theme.palette.info.light,
    color: theme.palette.info.dark,
}));

const StyledButton = styled(Button)(({ theme }) => ({
    whiteSpace: 'nowrap',
    minWidth: '8rem',
    marginLeft: theme.spacing(2),
}));

const StyledWarningIcon = styled(WarningAmber)(({ theme }) => ({
    color: theme.palette.warning.main,
}));

const StyledInfoIcon = styled(InfoOutlined)(({ theme }) => ({
    color: theme.palette.info.main,
}));

interface IInstanceStatusBarProps {
    instanceStatus: IInstanceStatus;
}

export const InstanceStatusBar = ({
    instanceStatus,
}: IInstanceStatusBarProps) => {
    if (trialHasExpired(instanceStatus)) {
        return <StatusBarExpired instanceStatus={instanceStatus} />;
    }

    if (trialExpiresSoon(instanceStatus)) {
        return <StatusBarExpiresSoon instanceStatus={instanceStatus} />;
    }

    if (isTrialInstance(instanceStatus)) {
        return <StatusBarExpiresLater instanceStatus={instanceStatus} />;
    }

    return null;
};

const StatusBarExpired = ({ instanceStatus }: IInstanceStatusBarProps) => {
    return (
        <StyledWarningBar data-testid={INSTANCE_STATUS_BAR_ID}>
            <StyledWarningIcon />
            <Typography sx={theme => ({ fontSize: theme.fontSizes.smallBody })}>
                <strong>Warning!</strong> Your free {instanceStatus.plan} trial
                has expired. <strong>Upgrade trial</strong> otherwise your{' '}
                <strong>account will be deleted.</strong>
            </Typography>
            <BillingLink />
        </StyledWarningBar>
    );
};

const StatusBarExpiresSoon = ({ instanceStatus }: IInstanceStatusBarProps) => {
    const timeRemaining = formatDistanceToNowStrict(
        parseISO(instanceStatus.trialExpiry!),
        { roundingMethod: 'floor' }
    );

    return (
        <StyledInfoBar data-testid={INSTANCE_STATUS_BAR_ID}>
            <StyledInfoIcon />
            <Typography sx={theme => ({ fontSize: theme.fontSizes.smallBody })}>
                <strong>Heads up!</strong> You have{' '}
                <strong>{timeRemaining}</strong> left of your free{' '}
                {instanceStatus.plan} trial.
            </Typography>
            <BillingLink />
        </StyledInfoBar>
    );
};

const StatusBarExpiresLater = ({ instanceStatus }: IInstanceStatusBarProps) => {
    return (
        <StyledInfoBar data-testid={INSTANCE_STATUS_BAR_ID}>
            <StyledInfoIcon />
            <Typography sx={theme => ({ fontSize: theme.fontSizes.smallBody })}>
                <strong>Heads up!</strong> You're currently on a free{' '}
                {instanceStatus.plan} trial account.
            </Typography>
            <BillingLink />
        </StyledInfoBar>
    );
};

const BillingLink = () => {
    const { hasAccess } = useContext(AccessContext);
    const navigate = useNavigate();

    if (!hasAccess(ADMIN)) {
        return null;
    }

    return (
        <StyledButton
            onClick={() => navigate('/admin/billing')}
            variant="outlined"
        >
            Upgrade trial
        </StyledButton>
    );
};
