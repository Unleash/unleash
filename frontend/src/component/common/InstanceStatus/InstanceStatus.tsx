import { useInstanceStatus } from 'hooks/api/getters/useInstanceStatus/useInstanceStatus';
import React, { FC, VFC, useEffect, useState, useContext } from 'react';
import { InstanceStatusBar } from 'component/common/InstanceStatus/InstanceStatusBar';
import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
import { Dialogue } from 'component/common/Dialogue/Dialogue';
import { Typography, useTheme } from '@mui/material';
import { useNavigate } from 'react-router-dom';
import { IInstanceStatus } from 'interfaces/instance';
import { ADMIN } from 'component/providers/AccessProvider/permissions';
import AccessContext from 'contexts/AccessContext';
import useInstanceStatusApi from 'hooks/api/actions/useInstanceStatusApi/useInstanceStatusApi';
import { trialHasExpired, canExtendTrial } from 'utils/instanceTrial';
import useToast from 'hooks/useToast';
import { formatUnknownError } from 'utils/formatUnknownError';

interface ITrialDialogProps {
    instanceStatus: IInstanceStatus;
    onExtendTrial: () => Promise<void>;
}

const TrialDialog: VFC<ITrialDialogProps> = ({
    instanceStatus,
    onExtendTrial,
}) => {
    const { hasAccess } = useContext(AccessContext);
    const navigate = useNavigate();
    const expired = trialHasExpired(instanceStatus);
    const [dialogOpen, setDialogOpen] = useState(expired);

    const onClose = (event: React.SyntheticEvent, muiCloseReason?: string) => {
        if (!muiCloseReason) {
            setDialogOpen(false);
            if (canExtendTrial(instanceStatus)) {
                onExtendTrial().catch(console.error);
            }
        }
    };

    useEffect(() => {
        setDialogOpen(expired);
        const interval = setInterval(() => {
            setDialogOpen(expired);
        }, 60000);
        return () => clearInterval(interval);
    }, [expired]);

    if (hasAccess(ADMIN)) {
        return (
            <Dialogue
                open={dialogOpen}
                primaryButtonText="Upgrade trial"
                secondaryButtonText={
                    canExtendTrial(instanceStatus)
                        ? 'Extend trial (5 days)'
                        : 'Remind me later'
                }
                onClick={() => {
                    navigate('/admin/billing');
                    setDialogOpen(false);
                }}
                onClose={onClose}
                title={`Your free ${instanceStatus.plan} trial has expired!`}
            >
                <Typography>
                    <strong>Upgrade trial</strong> otherwise your{' '}
                    <strong>account will be deleted.</strong>
                </Typography>
            </Dialogue>
        );
    }

    return (
        <Dialogue
            open={dialogOpen}
            secondaryButtonText="Remind me later"
            onClose={() => {
                setDialogOpen(false);
            }}
            title={`Your free ${instanceStatus.plan} trial has expired!`}
        >
            <Typography>
                Please inform your admin to <strong>Upgrade trial</strong> or
                your <strong>account will be deleted.</strong>
            </Typography>
        </Dialogue>
    );
};

export const InstanceStatus: FC = ({ children }) => {
    const { instanceStatus, refetchInstanceStatus, isBilling } =
        useInstanceStatus();
    const { extendTrial } = useInstanceStatusApi();
    const { setToastApiError } = useToast();
    const theme = useTheme();

    const onExtendTrial = async () => {
        try {
            await extendTrial();
            await refetchInstanceStatus();
        } catch (error: unknown) {
            setToastApiError(formatUnknownError(error));
        }
    };

    return (
        <div
            style={{
                height: '100%',
                backgroundColor: theme.palette.background.paper,
            }}
        >
            <ConditionallyRender
                condition={isBilling && Boolean(instanceStatus)}
                show={() => (
                    <>
                        <InstanceStatusBarMemo
                            instanceStatus={instanceStatus!}
                        />
                        <TrialDialog
                            instanceStatus={instanceStatus!}
                            onExtendTrial={onExtendTrial}
                        />
                    </>
                )}
            />
            {children}
        </div>
    );
};

const InstanceStatusBarMemo = React.memo(InstanceStatusBar);
