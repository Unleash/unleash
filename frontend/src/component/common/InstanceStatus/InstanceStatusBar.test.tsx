import { InstanceStatusBar } from 'component/common/InstanceStatus/InstanceStatusBar';
import { InstancePlan, InstanceState } from 'interfaces/instance';
import { render } from 'utils/testRenderer';
import { screen } from '@testing-library/react';
import { addDays, subDays } from 'date-fns';
import { INSTANCE_STATUS_BAR_ID } from 'utils/testIds';
import { UNKNOWN_INSTANCE_STATUS } from 'hooks/api/getters/useInstanceStatus/useInstanceStatus';

test('InstanceStatusBar should be hidden by default', async () => {
    render(<InstanceStatusBar instanceStatus={UNKNOWN_INSTANCE_STATUS} />);

    expect(
        screen.queryByTestId(INSTANCE_STATUS_BAR_ID)
    ).not.toBeInTheDocument();
});

test('InstanceStatusBar should be hidden when state is active', async () => {
    render(
        <InstanceStatusBar
            instanceStatus={{
                plan: InstancePlan.PRO,
                state: InstanceState.ACTIVE,
            }}
        />
    );

    expect(
        screen.queryByTestId(INSTANCE_STATUS_BAR_ID)
    ).not.toBeInTheDocument();
});

test('InstanceStatusBar should warn when the trial is far from expired', async () => {
    render(
        <InstanceStatusBar
            instanceStatus={{
                plan: InstancePlan.PRO,
                state: InstanceState.TRIAL,
                trialExpiry: addDays(new Date(), 15).toISOString(),
            }}
        />
    );

    expect(screen.getByTestId(INSTANCE_STATUS_BAR_ID)).toBeInTheDocument();
    expect(await screen.findByTestId(INSTANCE_STATUS_BAR_ID)).toMatchSnapshot();
});

test('InstanceStatusBar should warn when the trial is about to expire', async () => {
    render(
        <InstanceStatusBar
            instanceStatus={{
                plan: InstancePlan.PRO,
                state: InstanceState.TRIAL,
                trialExpiry: addDays(new Date(), 5).toISOString(),
            }}
        />
    );

    expect(screen.getByTestId(INSTANCE_STATUS_BAR_ID)).toBeInTheDocument();
    expect(await screen.findByTestId(INSTANCE_STATUS_BAR_ID)).toMatchSnapshot();
});

test('InstanceStatusBar should warn when trialExpiry has passed', async () => {
    render(
        <InstanceStatusBar
            instanceStatus={{
                plan: InstancePlan.PRO,
                state: InstanceState.TRIAL,
                trialExpiry: subDays(new Date(), 1).toISOString(),
            }}
        />
    );

    expect(screen.getByTestId(INSTANCE_STATUS_BAR_ID)).toBeInTheDocument();
    expect(await screen.findByTestId(INSTANCE_STATUS_BAR_ID)).toMatchSnapshot();
});

test('InstanceStatusBar should warn when the trial has expired', async () => {
    render(
        <InstanceStatusBar
            instanceStatus={{
                plan: InstancePlan.PRO,
                state: InstanceState.EXPIRED,
            }}
        />
    );

    expect(screen.getByTestId(INSTANCE_STATUS_BAR_ID)).toBeInTheDocument();
    expect(await screen.findByTestId(INSTANCE_STATUS_BAR_ID)).toMatchSnapshot();
});

test('InstanceStatusBar should warn when the trial has churned', async () => {
    render(
        <InstanceStatusBar
            instanceStatus={{
                plan: InstancePlan.PRO,
                state: InstanceState.CHURNED,
            }}
        />
    );

    expect(screen.getByTestId(INSTANCE_STATUS_BAR_ID)).toBeInTheDocument();
    expect(await screen.findByTestId(INSTANCE_STATUS_BAR_ID)).toMatchSnapshot();
});
