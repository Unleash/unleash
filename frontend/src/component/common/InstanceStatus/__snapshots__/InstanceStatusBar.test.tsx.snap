// Vitest Snapshot v1

exports[`InstanceStatusBar should warn when the trial has churned 1`] = `
<aside
  class="css-jmsogz"
  data-testid="INSTANCE_STATUS_BAR_ID"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-prk1jy-MuiSvgIcon-root"
    data-testid="WarningAmberIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 5.99 19.53 19H4.47L12 5.99M12 2 1 21h22L12 2z"
    />
    <path
      d="M13 16h-2v2h2zm0-6h-2v5h2z"
    />
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-rviqjc-MuiTypography-root"
  >
    <strong>
      Warning!
    </strong>
     Your free 
    Pro
     trial has expired. 
    <strong>
      Upgrade trial
    </strong>
     otherwise your
     
    <strong>
      account will be deleted.
    </strong>
  </p>
</aside>
`;

exports[`InstanceStatusBar should warn when the trial has expired 1`] = `
<aside
  class="css-jmsogz"
  data-testid="INSTANCE_STATUS_BAR_ID"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-prk1jy-MuiSvgIcon-root"
    data-testid="WarningAmberIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 5.99 19.53 19H4.47L12 5.99M12 2 1 21h22L12 2z"
    />
    <path
      d="M13 16h-2v2h2zm0-6h-2v5h2z"
    />
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-rviqjc-MuiTypography-root"
  >
    <strong>
      Warning!
    </strong>
     Your free 
    Pro
     trial has expired. 
    <strong>
      Upgrade trial
    </strong>
     otherwise your
     
    <strong>
      account will be deleted.
    </strong>
  </p>
</aside>
`;

exports[`InstanceStatusBar should warn when the trial is about to expire 1`] = `
<aside
  class="css-yx2rkt"
  data-testid="INSTANCE_STATUS_BAR_ID"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-cle2im-MuiSvgIcon-root"
    data-testid="InfoOutlinedIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M11 7h2v2h-2zm0 4h2v6h-2zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
    />
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-rviqjc-MuiTypography-root"
  >
    <strong>
      Heads up!
    </strong>
     You have
     
    <strong>
      4 days
    </strong>
     left of your free
     
    Pro
     trial.
  </p>
</aside>
`;

exports[`InstanceStatusBar should warn when the trial is far from expired 1`] = `
<aside
  class="css-yx2rkt"
  data-testid="INSTANCE_STATUS_BAR_ID"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-cle2im-MuiSvgIcon-root"
    data-testid="InfoOutlinedIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M11 7h2v2h-2zm0 4h2v6h-2zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
    />
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-rviqjc-MuiTypography-root"
  >
    <strong>
      Heads up!
    </strong>
     You're currently on a free
     
    Pro
     trial account.
  </p>
</aside>
`;

exports[`InstanceStatusBar should warn when trialExpiry has passed 1`] = `
<aside
  class="css-jmsogz"
  data-testid="INSTANCE_STATUS_BAR_ID"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-prk1jy-MuiSvgIcon-root"
    data-testid="WarningAmberIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 5.99 19.53 19H4.47L12 5.99M12 2 1 21h22L12 2z"
    />
    <path
      d="M13 16h-2v2h2zm0-6h-2v5h2z"
    />
  </svg>
  <p
    class="MuiTypography-root MuiTypography-body1 css-rviqjc-MuiTypography-root"
  >
    <strong>
      Warning!
    </strong>
     Your free 
    Pro
     trial has expired. 
    <strong>
      Upgrade trial
    </strong>
     otherwise your
     
    <strong>
      account will be deleted.
    </strong>
  </p>
</aside>
`;
